src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java

```diff
@@ -98,74 +116,59 @@ public boolean accept(AuditEvent event) {
+            final Patch patch = new Patch();
+            patch.parse(is);
+            final List<? extends FileHeader> fileHeaders = patch.getFiles();
+            for (FileHeader fileHeader : fileHeaders) {
+                final String fileName = getFileName(fileHeader);
+                final List<List<Integer>> lineRangeList = getLineRange(fileHeader);
                 final SuppressionPatchFilterElement element =
                         new SuppressionPatchFilterElement(fileName, lineRangeList);
                 filters.addFilter(element);
             }
+            System.out.println();
         }
         catch (IOException ex) {
             throw new CheckstyleException("an error occurred when load patch file", ex);
         }
     }
 
-    /**
-     * To separate different files's diff contents when there are multiple files in patch file.
-     * @param originPatch List of String
-     * @return patchedList List of List of String
-     */
-    private List<List<String>> getPatchList(List<String> originPatch) {
-        final List<List<String>> patchedList = new ArrayList<>();
-        boolean flag = true;
-        List<String> singlePatched = new ArrayList<>();
-        for (String str : originPatch) {
-            if (str.startsWith("diff ")) {
-                if (flag) {
-                    flag = false;
-                }
-                else {
-                    patchedList.add(singlePatched);
-                    singlePatched = new ArrayList<>();
-                }
-            }
-            singlePatched.add(str);
-        }
-        patchedList.add(singlePatched);
-        return patchedList;
-    }
-
-    private String getFileName(List<String> singlePatch) {
-        String fileName = null;
-        for (String str : singlePatch) {
-            if (str.startsWith("+++ ")) {
-                fileName = str.split("\\s")[1];
-                fileName = fileName.replaceAll("^b/", "");
-                break;
-            }
-        }
-        return fileName;
+    private String getFileName(FileHeader fileHeader) {
+        return fileHeader.getNewPath();
     }
 
-    private List<List<Integer>> getLineRange(Patch<String> patch) {
+    private List<List<Integer>> getLineRange(FileHeader fileHeader) {
         final List<List<Integer>> lineRangeList = new ArrayList<>();
-        for (int i = 0; i < patch.getDeltas().size(); i++) {
-            final Chunk targetChunk = patch.getDeltas().get(i).getTarget();
-            final List<Integer> lineRange = new ArrayList<>();
-            lineRange.add(targetChunk.getPosition());
-            lineRange.add(targetChunk.getPosition() + targetChunk.getLines().size());
-            lineRangeList.add(lineRange);
+        if (!"RENAME".equals(fileHeader.getChangeType().name())) {
+            for (HunkHeader hunkHeader : fileHeader.getHunks()) {
+                final EditList edits = hunkHeader.toEditList();
+                for (int i = 0; i < edits.size(); i++) {
```

Comment:
can we avoid `i` at all ? `for (Edit edit: edits)`

Reply:
done